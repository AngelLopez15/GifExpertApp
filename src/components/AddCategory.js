import React, {useState} from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {
    const [inputValue, setInputValue] = useState('')

    const handleInputChange= (e) => {
        setInputValue(e.target.value)
    }
    const handeleSubmit = (e) => {
        e.preventDefault()
        if(inputValue.trim().length>2){
            setCategories(categories => [inputValue, ...categories])
            setInputValue('')
        }
    }
    // como el form ya agrupa los elementos podemos quitar el fragment
    return (
        <form onSubmit={handeleSubmit}>
            <input 
                type="text"
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )
}

AddCategory.prototype = {
    setCategories: PropTypes.func.isRequired
}