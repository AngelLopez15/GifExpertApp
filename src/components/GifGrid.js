import React, {Fragment} from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem'


export const GifGrid = ({category}) => {
    // con el ":" renombramos el nombre data por images para una mejor comprension 
    const {data:images, loading} = useFetchGifs(category)

    return (
        <Fragment>
            <h3 className="animate__animated animate__backInLeft">{category}</h3>
            {/* se ocupa un perador logico && para evitar el ternario
            ya que si es false en el teernario hay que null y con el &&
            solo no ponemos nada si es false */}
            {loading && <p className="animate__animated animate__flash">loading</p> }
            <div className="card-grid">
                {
                    images.map(img =>(
                        <GifGridItem 
                            key={img.id}
                            {...img}
                        />
                    ))
                }
            </div>
        </Fragment>
    )
}
