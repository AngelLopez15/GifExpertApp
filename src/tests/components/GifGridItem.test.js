import React from 'react'

import { shallow } from "enzyme"
import  { GifGridItem } from "../../components/GifGridItem"

describe('Pruebas en <GifGridItem />', () => {
    const title='titulo'
    const url = 'https://localhost/imagen.jpg'

    test('Debe mostar el componente correctamente', () => {
        const wrapper= shallow(<GifGridItem title={title} url={url} />)
        expect(wrapper).toMatchSnapshot()
    })
    
})
