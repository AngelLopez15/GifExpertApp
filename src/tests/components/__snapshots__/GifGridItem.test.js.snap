// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe mostar el componente correctamente 1`] = `
<div>
  <div
    className="card animate__animated animate__fadeInUp"
  >
    <img
      alt="titulo"
      src="https://localhost/imagen.jpg"
    />
    <p>
      titulo
    </p>
  </div>
</div>
`;
